variables.tf

```diff
@@ -56,10 +56,17 @@ variable "transit_gateway_id" {
 variable "ram_resource_association_id" {
   description = "Identifier of the Resource Access Manager Resource Association"
   type        = string
+  default     = ""
 }
 
 variable "subnet_name_keyword_selector" {
   description = "Keyword matching the name of the subnet(s) for which the routing will be added (i.e. private)"
   type        = string
   default     = "private"
 }
+
+variable "transit_gateway_hub_name" {
+  description = "Name of the Transit Gateway to attach to"
+  type        = string
+  default     = ""
```

Comment:
Shouldn't we make this mandatory, otherwise it can be forgotten and left empty?

Reply:
This is optional. Either `ram_resource_association_id` or `transit_gateway_hub_name` is defined.

Reply:
As the PR name suggests, we are introducing an additional way to establish the attachment of the VPC with the TGW. We still support passing the IDs of the resources, explicitly.
Therefore, we need to allow both options to coexist. We could make use of the TF experimental variable validation feature, to ensure that we don't have _both_ variables set.